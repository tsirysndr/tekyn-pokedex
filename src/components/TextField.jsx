import React from "react";
import { View } from "react-native";

export default function TextField(props) {
  return <View></View>;
}