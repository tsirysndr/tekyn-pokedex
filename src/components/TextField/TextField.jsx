import styled from "@emotion/native";
import React from "react";
import { View } from "react-native";
import { useFonts, Roboto_700Bold } from "@expo-google-fonts/roboto";
import { useController } from "react-hook-form";

const TextInput = styled.TextInput`
  font-family: "Roboto_700Bold";
  background-color: #fff;
  height: 50px;
  width: 250px;
  border-radius: 4px;
  color: #000;
  padding: 10px;
`;

export default function TextField({ name, control }) {
 const { field } = useController({
    control,
    defaultValue: '',
    name,
  });
  const [fontsLoaded] = useFonts({ Roboto_700Bold });
  if (!fontsLoaded) {
    return null;
  }
  return (
    <View>
      <TextInput value={field.value} onChangeText={field.onChange} />
    </View>
  );
}
