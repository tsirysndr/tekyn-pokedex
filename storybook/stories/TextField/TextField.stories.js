import React from "react";
import TextField from "@Components/TextField";
import { storiesOf } from "@storybook/react-native";
import CenterView from "../CenterView";

storiesOf("TextField", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("default", () => <TextField />);