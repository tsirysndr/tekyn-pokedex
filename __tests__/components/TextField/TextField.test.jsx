import TextField from "@Components/TextField";
import { render } from "@testing-library/react-native";
import { useForm } from "react-hook-form";

test("renders TextField", () => {
  const TextFieldWrapper = () => {
    const { control } = useForm();
    return <TextField control={control} name="nameOrId"/>
  }
  const tree = render(
    <TextFieldWrapper />
  );
  expect(tree).toMatchSnapshot();
});
