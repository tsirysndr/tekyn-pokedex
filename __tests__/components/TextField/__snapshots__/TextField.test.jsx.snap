// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextField 1`] = `null`;
